### PR TITLE
Refactor some code to be more maintainable

### PR DESCRIPTION
**Pull request by technetos**
This PR moves some things around.  Additionally it switches from a custom `Result` type to 2 custom error types.  

Also:
+ Moves the ban thread that was overloaded with additonal things into its own module where we can just list functions to be run every hour, like a series of jobs
+ Moves command history into its own module
+ A small amount of renaming

### CODE REVIEW

File: src/jobs.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    thread::sleep,
+    time::Duration,
+};
+
+static JOBS_THREAD_INITIALIZED: AtomicBool = AtomicBool::new(false);
+
+pub(crate) fn start_jobs(cx: Context) {
+    if !JOBS_THREAD_INITIALIZED.load(Ordering::SeqCst) {
+        JOBS_THREAD_INITIALIZED.store(true, Ordering::SeqCst);
+        std::thread::spawn(move || -> Result<(), SendSyncError> {
```
Comment (khionu) [inline src/jobs.rs:14]: Why not spawn a Task? Would allow for async operations, and reusing a thread.

**Review by khionu — CHANGES_REQUESTED** [review]

File: src/jobs.rs (hunk 1 above)
Comment (technetos) [reply src/jobs.rs:14]: Because right now we are not using an async runtime

Comment (khionu) [reply src/jobs.rs:14]: Oh... right, we were waiting for a Serenity update, which might have released? Haven't checked their change logs, but they are importing Tokio https://github.com/serenity-rs/serenity/blob/current/Cargo.toml#L80.

**Review by khionu — APPROVED** [review]

File: src/commands.rs [hunk 2]
```diff
@@ -8,7 +8,7 @@ use reqwest::blocking::Client as HttpClient;
 use serenity::{model::channel::Message, prelude::Context};
 use std::{collections::HashMap, sync::Arc};
 
-const PREFIX: &'static str = "?";
+pub(crate) const PREFIX: &'static str = "?";
```
Comment (khionu) [inline src/commands.rs:11]: This should be a configuration at some point. Is this in scope for this PR?

Comment (technetos) [reply src/commands.rs:11]: Out of scope, but I dont disagree entirely.

**Review by khionu — APPROVED** [review]
This has inspired me to open a few issues, but otherwise LGTM
